packages/core/contracts/data-verification-mechanism/interfaces/SlashingLibraryInterface.sol

```diff
@@ -46,13 +46,16 @@ interface SlashingLibraryInterface {
      * @param totalStaked The total amount of tokens staked.
      * @param totalVotes The total amount of votes.
      * @param totalCorrectVotes The total amount of correct votes.
+     * @param priceRequestIndex The price request index within the resolvedPriceRequestIds array.
+     * @param isGovernance Whether the request is a governance request.
      * @return wrongVoteSlashPerToken The amount of tokens to slash for voting wrong.
      * @return noVoteSlashPerToken The amount of tokens to slash for not voting.
      */
     function calcSlashing(
         uint256 totalStaked,
         uint256 totalVotes,
         uint256 totalCorrectVotes,
+        uint256 priceRequestIndex,
         bool isGovernance
-    ) external pure returns (uint256 wrongVoteSlashPerToken, uint256 noVoteSlashPerToken);
+    ) external view returns (uint256 wrongVoteSlashPerToken, uint256 noVoteSlashPerToken);
```

Comment:
Needs to be `view` as the function reads from the storage. 

However I haven't modified the modifier in [SlashingLibrary](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/data-verification-mechanism/implementation/SlashingLibrary.sol#L45) although it seems not to be necessary. My interpretation is that as long as the interface is well defined there is no problem. Do you think I should do calcSlashing also view in SlashingLibrary?